e2e/tests/performance/memleak-imagery.perf.spec.js

```diff
@@ -1,121 +1,131 @@
+    await forceGC(page);
+    await captureHeapSnapshot(page, snapshotPath + 's1.heapsnapshot');
+
+    // Get and compare JSHeapUsedSize at different points in the test
+    let heapSize = await getHeapSize(page);
+    console.log(`Initial JSHeapUsedSize: ${heapSize}`);
+
+    await page.locator('a:has-text("My Items")').click({ button: 'right' });
     await page.locator('text=Import from JSON').click();
-
-    // Upload Performance Display Layout.json
     await page.setInputFiles('#fileElem', filePath);
-
-    // Click text=OK
     await page.locator('text=OK').click();
-
     await expect(
       page.locator('a:has-text("Performance Display Layout Display Layout")')
     ).toBeVisible();
   });
 
-  test('Embedded View Large for Imagery is performant in Fixed Time', async ({ page, browser }) => {
-    await page.goto('./', { waitUntil: 'networkidle' });
-
-    // To to Search Available after Launch
+  test('Embedded View Large for Imagery is performant in Fixed Time', async ({ page }) => {
     await page.locator('[aria-label="OpenMCT Search"] input[type="search"]').click();
-    // Fill Search input
     await page
       .locator('[aria-label="OpenMCT Search"] input[type="search"]')
       .fill('Performance Display Layout');
-    //Search Result Appears and is clicked
     await Promise.all([
       page.waitForNavigation(),
       page.locator('a:has-text("Performance Display Layout")').first().click()
     ]);
 
-    //Time to Example Imagery Frame loads within Display Layout
     await page.waitForSelector('.c-imagery__main-image__bg', { state: 'visible' });
-    //Time to Example Imagery object loads
     await page.waitForSelector('.c-imagery__main-image__background-image', { state: 'visible' });
+    await page.waitForTimeout(3 * 1000);
+    await forceGC(page);
+    await captureHeapSnapshot(page, snapshotPath + 's2.heapsnapshot');
+    // Get and compare JSHeapUsedSize at different points in the test
+    let heapSize = await getHeapSize(page);
+    console.log(`second JSHeapUsedSize: ${heapSize}`);
+  });
 
-    const client = await page.context().newCDPSession(page);
-    await client.send('HeapProfiler.enable');
-    await client.send('HeapProfiler.startSampling');
-    // await client.send('HeapProfiler.collectGarbage');
-    await client.send('Performance.enable');
+  test.afterEach(async ({ page }) => {
+    await page.goto(
+      './#/browse/mine?tc.mode=local&tc.timeSystem=utc&view=grid&tc.startDelta=1800000&tc.endDelta=30000',
+      { waitUntil: 'domcontentloaded' }
+    );
+    await page.waitForTimeout(3 * 1000);
+    await forceGC(page);
+    await captureHeapSnapshot(page, snapshotPath + 's3.heapsnapshot');
+    // Get and compare JSHeapUsedSize at different points in the test
+    let heapSize = await getHeapSize(page);
+    console.log(`Final JSHeapUsedSize: ${heapSize}`);
+
+    console.log('Heap snapshot captured');
+    const reader = BrowserInteractionResultReader.from(snapshotPath);
```

Comment:
To use MemLab the directory needs to have this structure:

```
/path/to/dump/
└── data
    └── cur
        ├── s1.heapsnapshot     # heap snapshot after the url callback (initial page load)
        ├── s2.heapsnapshot     # heap snapshot after the action callback (after target interaction)
        ├── s3.heapsnapshot     # heap snapshot after the back callback (after reverting target interaction)
        └── snap-seq.json       # meta data about each browser interaction step
```

In this case, `snap-seq.json` could just be a static file with the following content:
```
[
  {
    "name": "page-load", 
    "snapshot": true,
    "type": "baseline",
    "idx": 1,
    "JSHeapUsedSize": 0
  },
  {
    "name": "action-on-page",
    "snapshot": true,
    "type": "target",
    "idx": 2,
    "JSHeapUsedSize": 0
  },
  {
    "name": "revert",
    "snapshot": true,
    "type": "final",
    "idx": 3,
    "JSHeapUsedSize": 0
  }
]
```

For more details about the directory structure and the metadata file, please checkout the discussion here:
https://github.com/facebook/memlab/issues/35